File: src/main/java/flux/app/api/session/AuthController.java
package flux.app.api.session;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import flux.app.api.models.UserModel;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/session")
public class AuthController {
  private final AuthService authService;

  @Autowired
  AuthController(AuthService authService) {
    this.authService = authService;
  }

  @PostMapping("auth")
  Mono<UserModel> call() {
    Mono<UserModel> user = authService.call();

    return user;
  }
}

File: src/main/kotlin/flux/app/api/session/AuthController.kt
package flux.app.api.session

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
@RequestMapping("api/session")
class AuthCotroller (private val authService: AuthService) {
  @PostMapping("auth")
  fun call(): Mono<ResponseEntity<String>> {

    authService.call()

    return Mono.just(ResponseEntity("HELLO!", HttpStatus.OK))
  }
}
